bundle/edu.gemini.pot/src/main/java/edu/gemini/pot/sp/memImpl/MemObservation.java

```diff
@@ -109,7 +109,13 @@ public int getObservationNumber() {
     }
 
     public void setObservationNumber(int number) {
-        _obsNumber = number;
+        getProgramWriteLock();
+        try {
+            _obsNumber = number;
+            ((ProgramData) getProgram().getDocumentData()).ensureMaxEqualToOrGreaterThan(number);
+        } finally {
+            returnProgramWriteLock();
+        }
```

Comment:
This is the actual bug fix.  Setting observation numbers explicitly after they are created is only done by the merge process (except for test cases).